File: java/P330.java
class Solution {
    public int minPatches(int[] nums, int n) {
        int answer = 0;
        long max = 0;
        int i = 0;
        while (max<n) {
            if (i >= nums.length || nums[i]-1 > max) {
                answer++;
                max += max+1;
            } else {
                max += nums[i];
                i++;
            }
        }
        return answer;
    }
}

File: python/P330.py
class Solution(object):
    def minPatches(self, nums, n):
        max_num = 0
        count = 0
        i = 0
        while max_num < n:
            if i >= len(nums) or max_num < nums[i]-1:
                max_num += max_num+1
                count += 1
            else:
                max_num += nums[i]
                i += 1;
        return count